backend/generalhw.pm

```diff
@@ -45,20 +45,20 @@ sub run_cmd ($self, $cmd, @extra_args) {
 
     push @full_cmd, @extra_args;
 
-    my ($stdin, $stdout, $stderr, $ret);
-
+    bmwqemu::diag("Calling $cmd");
+    my ($p, $out);
     {
         # Do not let the SIGCHLD handler of Mojo::IOLoop::ReadWriteProcess::Session steal the exit code from IPC::Run
         local $SIG{CHLD};
-        eval { $ret = IPC::Run::run(\@full_cmd, \$stdin, \$stdout, \$stderr) };
-        die "Unable to run command '@full_cmd' (deduced from test variable $cmd): $@\n" if $@;
+        $p = IPC::Run::start(\@full_cmd, '&>', \$out);
     }
-    chomp $stdout;
-    chomp $stderr;
-
-    die "$cmd: stdout: $stdout, stderr: $stderr" unless $ret;
-    bmwqemu::diag("$cmd: stdout: $stdout, stderr: $stderr");
-    return $stdout;
+    while ($p->pumpable) {
+        # stream the cmd output to the live log
+        $p->pump();
+        print($out);
+        $out = '';
+    }
```

Comment:
After reading the documentation of IPC::Run I *suppose* this works and is *not* a spinning loop. However, I'm still wondering whether just invoking `system` (with proper error handling) would just do the trick here as well. After all, all this does is immediately forwarding the output as-is. This seems as good as just invoking the process via `system`.

Reply:
I would be fine with that too.